unit_test.py

```diff
@@ -411,7 +459,121 @@ def test_services_table(self):
         # Assert
         self.assertDictEqual(result, expected)
 
-        
-
+    def test_bgp_as_detection(self):
+        """Test that the correct bgp as is detected"""
+        """Calico is tested by default in all thre previous tests"""
+        """Raise a Calico Exception, that triggers testing for Kube-Router"""
+        with patch('kubernetes.client.CustomObjectsApi.get_cluster_custom_object', MagicMock(side_effect=Exception("Test"))):
+            build = VkaciBuilTopology(
+                VkaciEnvVariables(self.vars), ApicMethodsMock())
+            build.update()
+            asn = build.get_cluster_as()
+            self.assertEqual(asn, '56001')
+        CiliumBGPPeeringPolicies = json.loads("""

```

Comment:
Including an entire Cilium policy into the unit test was not required since mocking only what is required and not everything is best practice for unit testing.